api-gateway-routes/api_gateway_proxy_function.rb

```diff
@@ -50,11 +50,44 @@ def on_connect(event, context)
     :queueName => queue_name,
     :executionType => authorizer['execution_type']
   }
-  response = lambda_client.invoke({
-    function_name: ENV['BUILD_AND_RUN_PROJECT_LAMBDA_ARN'] || 'javaBuilderExecuteCode:13',
-    invocation_type: 'Event',
-    payload: JSON.generate(payload)
-  })
+
+  response = nil
+  if authorizer['mini_app_type'] == 'neighborhood'
+    response = lambda_client.invoke({
+      function_name: ENV['BUILD_AND_RUN_NEIGHBORHOOD_PROJECT_LAMBDA_ARN'],
+      invocation_type: 'Event',
+      payload: JSON.generate(payload)
+    })
+  elsif authorizer['mini_app_type'] == 'console'
+    response = lambda_client.invoke({
+      function_name: ENV['BUILD_AND_RUN_CONSOLE_PROJECT_LAMBDA_ARN'],
+      invocation_type: 'Event',
+      payload: JSON.generate(payload)
+    })
+  elsif authorizer['mini_app_type'] == 'theater'
+    response = lambda_client.invoke({
+      function_name: ENV['BUILD_AND_RUN_THEATER_PROJECT_LAMBDA_ARN'],
+      invocation_type: 'Event',
+      payload: JSON.generate(payload)
+    })
+  elsif authorizer['mini_app_type'] == 'playground'
+    response = lambda_client.invoke({
+      function_name: ENV['BUILD_AND_RUN_PLAYGROUND_PROJECT_LAMBDA_ARN'],
+      invocation_type: 'Event',
+      payload: JSON.generate(payload)
+    })
```

Comment:
Seeing this much duplicated code makes me wonder, as a reader, if `invocation_type` and `payload` are also different based on the app type. Since we're just changing the function name, it might be clearer to set a `function_name` local variable in the if block, or pluck the correct one out of a `function_names` hash. There's some error handling to add for the unknown name scenario, but something like this?

```ruby
# build the function_names hash, mapping names to ARN's
# check to see if the mini_app_type is not in the hash, use 'playground' or throw 400

response = lambda_client.invoke({
  function_name: function_names[authorizer['mini_app_type'],
  invocation_type: 'Event',
  payload: JSON.generate(payload)
})
```